packages/node/src/index.ts

```diff
@@ -200,23 +201,48 @@ async function compile(
         // null represents a not found
         if (cached === null) return null;
         try {
+          let entry: File | undefined;
           let source: string | Buffer = readFileSync(fsPath);
+
+          const { mode } = lstatSync(fsPath);
+          if (isSymbolicLink(mode)) {
+            entry = new FileFsRef({ fsPath, mode });
+          }
+
+          if (isEdgeFunction && basename(fsPath) === 'package.json') {
+            // For Edge Functions, patch "main" field to prefer "browser" or "module"
+            const pkgJson = JSON.parse(source.toString());
+            for (const prop of ['browser', 'module']) {
+              const val = pkgJson[prop];
+              if (typeof val === 'string') {
+                debug(`Using "${prop}" field in ${fsPath}`);
+                pkgJson.main = val;
+
+                // Create the `entry` with the original so that the output is unmodified
+                if (!entry) {
+                  entry = new FileBlob({ data: source, mode });
+                }
+
+                // Return the modified `package.json` to nft
+                source = JSON.stringify(pkgJson);
+                break;
+              }
+            }
+          }
+
           if (
             (fsPath.endsWith('.ts') && !fsPath.endsWith('.d.ts')) ||
             fsPath.endsWith('.tsx')
           ) {
             source = compileTypeScript(fsPath, source.toString());
           }
-          const { mode } = lstatSync(fsPath);
-          let entry: File;
-          if (isSymbolicLink(mode)) {
-            entry = new FileFsRef({ fsPath, mode });
-          } else {
+
+          if (!entry) {
             entry = new FileBlob({ data: source, mode });
           }
           fsCache.set(relPath, entry);
           sourceCache.set(relPath, source);
-          return source.toString();
+          return source;
```

Comment:
This can be a buffer according to line 205 so I think you need to keep the tostring()